accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,550 @@
+NuGet should use this next syntax to automatically understand that `net5.0` can
+be consumed from `net6.0-win` (but not the other way around). But more
+importantly, this notation will also enable developers to intuitively understand
+compatibility relationships because they are expressed by naming, rather than by
+mapping tables. Yay!
+
+## Scenarios and User Experience
+
+### Vary implementation
+
+Ida is working on a Xamarin Forms application that supports Android, iOS, and
+Windows. Her application needs GPS information, but only a very limited set.
+Since there is no portable GPS API, she writes her own little abstraction
+library using multi-targeting.
+
+By doing so, she's able to encapsulate the GPS access without having to
+multi-target her entire application, just this one area.
+
+```C#
+public static class GpsLocation
+{
+    public static bool IsSupported
+    {
+        get
+        {
+#if ANDROID || IOS || WINDOWS
+            return true;
+#else
+            return false;
+#endif
+        }
+    }
+
+    public static (double Latitude, double Longitude) GetCoordinates()
+    {
+#if ANDROID
+        return AndroidAPI();
+#elif IOS
+        return AppleAPI();
+#elif WINDOWS
+        return WindowsAPI();
+#else
+        throw new PlatformNotSupportedException();
+#endif
+    }
+}
+```
+
+### Vary API surface
+
+Ada is a developer on SkiaSharp, a cross-platform 2D graphics API for .NET
+platforms based on Google's Skia Graphics Library. The project is already using
+multi-targeting to provide different implementations for different platform. To
+make it easier to use, she's adding a new `SkiaSharpImage` type, which
+represents a bitmap and is constructed via OS-provided data types. Ada uses
+`#if` to expose different constructors on different platforms:
+
+```C#
+public static class SkiaSharpImage
+{
+#if ANDROID
+    public SkiaSharpImage(Android.Media.Image nativeImage) { /* ... */  }
+#endif
+
+#if IOS
+    public SkiaSharpImage(NSImage nativeImage) { /* ... */ }
+#endif
+
+#if WINDOWS
+    public SkiaSharpImage(Windows.Media.BitmapImage nativeImage) { /* ... */ }
+#endif
+}
+```
+
+## Requirements
+
+### Goals
+
+* Use naming that aligns with product strategy
+* Merge .NET Core and .NET Standard into a single concept
+* Developers should be able to understand compatibility relationships without
+  having to consult a mapping table.
+* Provide compatibility with existing concepts and NuGet packages
+* It would be great to get this into .NET 5 Preview 1
+
+### Non-Goals
+
+* Replace TFMs or expand runtime identifiers (RIDs)
+* Support multi-targeting between OS versions
+
+## Design
+
+| TFM             | Compatible With                                            | Comments                          |
+|-----------------|------------------------------------------------------------|-----------------------------------|
+| net5.0          | net1..4 (with NU1701 warning)                              | No WinForms or WPF                |
```

Comment:
I'd imagine we want this warning for net5.0 -> netcoreapp3.1 w/ frameworkrefs to also apply for most "profiles" as well. Except, when net5.0-windows is used, i'd imagine we don't want that warning.

Reply:
Hmm... I think `netcoreapp3.1` w/ the desktop framework ref should be a warning for `net5.0` and fail for all OS flavors except `net5.0-windows`. If the project says a specific operating system that isn't Windows, then it will for sure not work. If it's `net5.0` then it can work if the code only runs on Windows. Does this make sense?